sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -89,6 +89,10 @@ class SimpleFunctionRegistry extends FunctionRegistry {
     functionBuilders.iterator.map(_._1).toList.sorted
   }
 
+  private[catalyst] def functionSet(): Set[String] = synchronized {
+    functionBuilders.iterator.map(_._1).toSet
```

Comment:
Seems we are still creating the set every time when we call `FunctionRegistry.builtin.functionSet`. Can you create this set in the object of `FunctionRegistry`?